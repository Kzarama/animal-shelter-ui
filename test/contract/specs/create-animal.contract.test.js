import { AnimalController } from "../../../controllers";
import { provider } from "../config/initPact";

const animal = {
  name: "manchas",
  breed: "Bengali",
  gender: "Female",
  vaccinated: true,
};

describe("Animal Service", () => {
  beforeAll(async () => {
    await provider.setup();
  });

  describe("When a request to create an animal is made", () => {
    beforeAll(async () => {
      await provider.addInteraction({
        uponReceiving: "a request to create an animal",
        state: "create animal",
        withRequest: {
          method: "POST",
          path: "/animals",
          body: animal,
        },
        willRespondWith: {
          status: 201,
        },
      });
    });

    test("should return the correct data", async () => {
      const response = await AnimalController.register(animal);
      expect(response.data).toMatchSnapshot();

      await provider.verify();
    });
  });
  afterAll(async () => provider.finalize());
});
