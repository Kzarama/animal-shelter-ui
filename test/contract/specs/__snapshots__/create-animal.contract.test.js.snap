// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Service When a request to create an animal is made should return the correct data 1`] = `""`;
